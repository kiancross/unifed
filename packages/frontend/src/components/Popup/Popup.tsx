/*
 * CS3099 Group A3
 */

import { Snackbar } from "@material-ui/core";
import { Alert } from "@material-ui/lab";
import { ReactElement } from "react";

/**
 * Properties for the [[`Popup`]] component.
 */
export interface PopupProps {
  /**
   * Message to display in the popup.
   */
  message?: string;
}

/**
 * Displays an error to the user in popup form
 *
 * Outline:
 *
 *  - Used if the user has tried to take an action that is invalid e.g. enterring the wrong login information.
 *
 * @param props Properties passed to the component. See [[`PopupProps`]].
 *
 * @internal
 */
export function Popup(props: PopupProps): ReactElement {
  return (
    <Snackbar anchorOrigin={{ vertical: "top", horizontal: "center" }} open={!!props.message}>
      <Alert severity="error">{props.message}</Alert>
    </Snackbar>
  );
}
