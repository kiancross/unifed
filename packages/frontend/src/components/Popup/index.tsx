/*
 * CS3099 Group A3
 */

import Popup from "./Popup";
export default Popup;
