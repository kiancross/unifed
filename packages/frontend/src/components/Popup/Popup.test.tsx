/*
 * CS3099 Group A3
 */

import React from "react";
import Popup from "./Popup";
import renderer from "react-test-renderer";

test("Render", () => {
  const component = renderer.create(<Popup />);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
