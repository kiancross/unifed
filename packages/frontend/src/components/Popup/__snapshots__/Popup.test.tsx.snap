// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render 1`] = `
<div>
  <h1 />
  <button>
    close me
  </button>
</div>
`;
